crates/sui-graphql-rpc/src/extensions/limits_info.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::{
+    extensions::{Extension, ExtensionContext, ExtensionFactory, NextRequest, NextValidation},
+    value, Response, ServerError, ValidationResult,
+};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+use crate::config::ServiceConfig;
+
+pub struct LimitsInfo;
+
+impl ExtensionFactory for LimitsInfo {
+    fn create(&self) -> Arc<dyn Extension> {
+        Arc::new(LimitsInfoExtension::default())
+    }
+}
+
+#[derive(Default)]
+struct LimitsInfoExtension {
+    validation_result: Mutex<Option<ValidationResult>>,
+}
+
+#[async_trait::async_trait]
+impl Extension for LimitsInfoExtension {
+    async fn request(&self, ctx: &ExtensionContext<'_>, next: NextRequest<'_>) -> Response {
```

Comment:
I'm not sure it makes sense to do this for all responses -- this should be an opt-in feature, maybe based on the inclusion of some GET parameter or something?